### DISCUSSION TITLE
Can I use Pactum to intercept HTTP requests ?

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#278] [General]
Can I use Pactum to intercept HTTP requests ?

I have to test some interactions with external applications. I was hoping if I can intercept those HTTP requests to the external application and send my response back, then I can test my application services easily.
I was wondering whether pactum has that capability or not, I could not find anything similar in the documentation. 
Hopefully, the community can provide an answer.

Thanks, 
Mihir

**Comment by ASaiAnudeep** [comment]
It sounds like [component testing](https://pactumjs.github.io/guides/component-testing.html). Please check the docs.

**Comment by ian-mj** [comment]
I have a similar need to Mihir, have read the documentation (including component testing, mock servers, interaction handlers, ...), but still could not get it to work for my use case.

My application receives a POST request, and in the background makes a POST request to an external service. I am trying to mock a response from that 2nd call, to validate the logic of the rest of my application. But that part does not seem to work.

Is this something Pactum can do?
